FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.JobManagement/AzureStorageQueueClient.cs

```diff
@@ -0,0 +1,682 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using Azure;
+using Azure.Data.Tables;
+using Azure.Identity;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Exceptions;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Extensions;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Models;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Models.AzureStorage;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Synapse.JobManagement
+{
+    public class AzureStorageQueueClient<TJobInfo> : IQueueClient
+        where TJobInfo : AzureStorageJobInfo, new()
+    {
+        private readonly TableClient _azureJobInfoTableClient;
+
+        private readonly QueueClient _azureJobMessageQueueClient;
+
+        private readonly ILogger<AzureStorageQueueClient<TJobInfo>> _logger;
+
+        /// <summary>
+        /// The maximum size of a single entity, including all property values is 1 MiB,
+        /// see https://docs.microsoft.com/en-us/azure/storage/tables/scalability-targets#scale-targets-for-table-storage.
+        /// The definition and result are serialized string of objects,
+        /// we should be careful that should not contain large fields when define the definition and result class.
+        /// Add an assert here to make sure the size of definition and result aren't larger than 1MiB.
+        /// </summary>
+        private const int MaximumSizeOfAnEntityInBytes = 1024 * 1024;
+
+        public AzureStorageQueueClient(
+            IStorage storage,
```

Comment:
We need a factory to create queue and table client, instead of create it here for different auth. 

In PaaS, we need to have special logic to create through MI.

Reply:
Updated.